finagle-redis/src/main/scala/com/twitter/finagle/Redis.scala

```diff
@@ -63,4 +66,48 @@ object Redis extends Client[Command, Reply] {
 
   def newService(dest: Name, label: String): Service[Command, Reply] =
     client.newService(dest, label)
+
+  object Subscribe extends com.twitter.finagle.Client[SubscribeCommand, Unit] {
+
+    object SubscribeClient {
+      /**
+       * Default stack parameters used for redis client.
+       */
+      val defaultParams: Stack.Params = StackClient.defaultParams +
+        param.ProtocolLibrary("redis.subscribe")
+
+      /**
+       * A default client stack which supports the pipelined redis client.
+       */
+      def newStack: Stack[ServiceFactory[SubscribeCommand, Unit]] = StackClient.newStack
+        .replace(DefaultPool.Role, SingletonPool.module[SubscribeCommand, Unit])
+    }
+
+    case class SubscribeClient(
```

Comment:
In that case, we'll have to keep track of the client state, and prevent users from issuing ordinary commands to a pub/sub client by throwing IllegalStateException. I think it's better to keep them separate for type safety.